dataprofiler/profilers/helpers/report_helpers.py

```diff
@@ -100,7 +100,54 @@ def _prepare_report(report, output_format=None, omit_keys=None):
         if isinstance(value, set):
             value = sorted(list(value))
 
-        if isinstance(value, dict):
+        if "*" in omit_keys:
+            continue
+
+        # For data_stats (in structured case), need to recurse through a list
+        elif key == "data_stats" and isinstance(value, list) \
+                and "data_stats" not in omit_keys:
+
+            fmt_report["data_stats"] = []
+
+            for col_ind in range(len(value)):
+                col_name = str(value[col_ind].get('column_name'))
+                # column is being omitted
+                if f"data_stats.{col_name}" in omit_keys \
+                        or "data_stats.*" in omit_keys:
+                    fmt_report["data_stats"].append(None)
+                    continue
```

Comment:
the issue becomes, these are technically already looping through omit keys. Here we have 2 `in` conditions. and then below we have a third loop through `in` with the `for`.  We also then have to remember that these conditions exist in combination with what we are doing below. Maybe it is best to move these into the for loop below for clarity:

```python
next_layer_omit_keys = []
for omit_key in omit_keys:
    
    # case where omit_key removes this column entirely
    if omit_key in {"*", col_name, f"data_stats.{col_name}":
        fmt_report["data_stats"].append(None)
        break
    
    # case where the omit extends into this column
    omit_key_split = omit_key.split('.', 1)
    if omit_key_split[0] in {"*", col_name}:
        next_layer_omit_keys.append(omit_key_split[1])
    
    fmt_report["data_stats"].append(
        _prepare_report(value[col_ind], output_format,
                        next_layer_omit_keys))
```

Reply:
Resolved, now we check this condition within the omit_key loop